### PR TITLE
Print parse error before usage statement

### PR DESCRIPTION
**Pull request by derekbekoe**
- Also, parse error now sent to logger instead of stderr directly so it is colorized also.

Closes https://github.com/Azure/azure-cli/issues/1472

BEFORE:
```
$ az vm list sdfsdf
usage: az [-h] [--output {json,tsv,list,table,jsonc}] [--verbose] [--debug]
          [--query JMESPATH]
          {sql,redis,network,batch,iot,account,group,vm,appservice,storage,feature,vmss,cloud,policy,keyvault,logout,acs,acr,container,role,tag,ad,provider,resource,context,component,feedback,taskhelp,login,configure}
          ...
az: error: unrecognized arguments: sdfsdf
```

AFTER:
```
$ az vm list sdfsdf
az: error: unrecognized arguments: sdfsdf
usage: az [-h] [--output {json,tsv,list,table,jsonc}] [--verbose] [--debug]
          [--query JMESPATH]
          {sql,redis,network,batch,iot,account,group,vm,appservice,storage,feature,vmss,cloud,policy,keyvault,logout,acs,acr,container,role,tag,ad,provider,resource,context,component,feedback,taskhelp,login,configure}
          ...
```

### CODE REVIEW

**Comment by derekbekoe** [comment]
cc: @squillace

File: src/azure-cli-core/azure/cli/core/parser.py [hunk 1]
```diff
@@ -3,15 +3,20 @@
...
 
+import argparse
 import argcomplete
 
 import azure.cli.core.telemetry as telemetry
 import azure.cli.core._help as _help
 from azure.cli.core._util import CLIError
 from azure.cli.core._pkg_util import handle_module_not_installed
 
+import azure.cli.core._logging as _logging
```
Comment (troydai) [inline src/azure-cli-core/azure/cli/core/parser.py]: We should make the `_logging` module public. It is referenced everywhere. It doesn't make sense keep it a private module.

File: src/azure-cli-core/azure/cli/core/parser.py [hunk 2]
```diff
@@ -142,7 +147,11 @@ def validation_error(self, message):
     def error(self, message):
         telemetry.set_user_fault('parse error: {}'.format(message))
         self._handle_command_package_error(message)
-        return super(AzCliCommandParser, self).error(message)
+
+        args = {'prog': self.prog, 'message': message}
+        logger.error('%(prog)s: error: %(message)s', args)
+        self.print_usage(sys.stderr)
+        self.exit(2)
```
Comment (troydai) [inline src/azure-cli-core/azure/cli/core/parser.py]: About this. Will exit here change the existing workflow (I have no idea, so I'm just asking.)

Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/parser.py]: By default, argparse prints usage, prints error message then exits with code 2.
I have switched around the order of print usage and print error.
So I override this method which exits with code 2 as well.
https://github.com/python/cpython/blob/master/Lib/argparse.py#L2378

File: src/azure-cli-core/azure/cli/core/parser.py (hunk 1 above)
Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/parser.py]: Noted. Will make this change in a separate PR.
